llvm/test/CodeGen/AMDGPU/lower-buffer-fat-pointers-control-flow.ll

```diff
@@ -352,3 +352,107 @@ loop.exit:
 exit:
   ret float %sum
 }
+
+;; Test that the optimization works correctly for phi nodes that repeat the same
```

Comment:
Not an optimization

Reply:
I'd characterize switching

```llvm
%phi.rsrc = phi [ %v.rsrc, %l1 ], [ %v.rsrc, %l2 ], ...
%phi.off = phi [ %v1.off, %l1 ], [ %v2.off, %l2 ], ...
```
(the natural lowering of `phi`)

to 
```llvm
%phi.rsrc = %v.rsrc
%phi.off = [ %v1.off, %l1 ], [ %v2.off, %l2 ]
```
as an optimization.